### PR TITLE
MODCON-73. Removed unnecessary folioExecution.getInstance() and setting context again in scope of method.

### PR DESCRIPTION
**Pull request by azizbekxm**
## Pre-Merge Checklist:
Before merging this PR, please go through the following list and take appropriate actions.

- Does this PR meet or exceed the expected quality standards?
  - [ ] Code coverage on new code is 80% or greater
  - [ ] Duplications on new code is 3% or less
  - [ ] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added or removed?
  - [ ] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [ ] Were permissions changed, added or removed?
  - [ ] Are there new interface dependencies?
  - [ ] There are no breaking changes in this PR.

If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all the appropriate links to blocked/related issues?
- Are the JIRAs under active development?
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?
- Did you modify code to call some additional endpoints?
  - [ ] If so, do you check that necessary module permission added in ModuleDescriptor-template.yaml file?

Ideally, all the PRs involved in breaking changes would be merged on the same day
to avoid breaking the folio-testing environment.
Communication is paramount if that is to be achieved,
especially as the number of inter-module and inter-team dependencies increase.

While it's helpful for reviewers to help identify potential problems,
ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: src/main/java/org/folio/consortia/service/impl/UserAffiliationServiceImpl.java [hunk 1]
```diff
@@ -100,10 +91,7 @@ public void deletePrimaryUserAffiliation(String eventPayload) {
       PrimaryAffiliationEvent affiliationEvent = createPrimaryAffiliationEvent(userEvent);
       String data = objectMapper.writeValueAsString(affiliationEvent);
 
-      // context is changed in deleteShadowUsers() method and context is empty after deleteShadowUsers() method, so we need to set context again.
-      try (var context = new FolioExecutionContextSetter(prepareContextForTenant(centralTenantId, folioModuleMetadata, currentContext))) {
-        kafkaService.send(KafkaService.Topic.CONSORTIUM_PRIMARY_AFFILIATION_DELETED, consortiaTenant.getConsortiumId().toString(), data);
```
Comment (azizbekxm) [inline src/main/java/org/folio/consortia/service/impl/UserAffiliationServiceImpl.java:105]: after resent changes in folio-spring-base we doesn't need to set back again context in method. it automatically back to initial context in scope of method.

Comment (azizbekxm) [reply src/main/java/org/folio/consortia/service/impl/UserAffiliationServiceImpl.java:105]: I tested manually I didn't see any errors

File: src/main/java/org/folio/consortia/service/impl/UserTenantServiceImpl.java [hunk 2]
```diff
@@ -109,7 +109,7 @@ public UserTenantCollection getByUserId(UUID consortiumId, UUID userId, Integer
   @Transactional
   public UserTenant save(UUID consortiumId, UserTenant userTenantDto, boolean isSystemUserContextRequired) {
     log.debug("Going to save user with id: {} into tenant: {}", userTenantDto.getUserId(), userTenantDto.getTenantId());
-    FolioExecutionContext currentTenantContext = (FolioExecutionContext) folioExecutionContext.getInstance();
+    FolioExecutionContext currentTenantContext = folioExecutionContext;
```
Comment (tatsiana-tarhonskaya) [inline src/main/java/org/folio/consortia/service/impl/UserTenantServiceImpl.java:112]: we should not need this variable, setting context can be done using folioExecutionContext field

**Review by SerhiiNosko — APPROVED** [review]

File: src/main/java/org/folio/consortia/service/impl/UserTenantServiceImpl.java [hunk 3]
```diff
@@ -123,10 +122,10 @@ public UserTenant save(UUID consortiumId, UserTenant userTenantDto, boolean isSy
     if (isSystemUserContextRequired) {
       createOrUpdateShadowUserWithSystemUserContext(userTenantDto.getUserId(), shadowUser, userTenantDto);
     } else {
-      createOrUpdateShadowUserWithRequestedContext(userTenantDto.getUserId(), shadowUser, userTenantDto, currentTenantContext);
+      createOrUpdateShadowUserWithRequestedContext(userTenantDto.getUserId(), shadowUser, userTenantDto, folioExecutionContext);
     }
 
-    try (var context = new FolioExecutionContextSetter(prepareContextForTenant(currentTenantId, folioModuleMetadata, currentTenantContext))) {
+    try (var context = new FolioExecutionContextSetter(prepareContextForTenant(currentTenantId, folioModuleMetadata, folioExecutionContext))) {
```
Comment (tatsiana-tarhonskaya) [inline src/main/java/org/folio/consortia/service/impl/UserTenantServiceImpl.java:128]: It looks like setting of the context with same tenantId as it already has, isn't it? Is it needed?

Comment (azizbekxm) [reply src/main/java/org/folio/consortia/service/impl/UserTenantServiceImpl.java]: I checked, it turned out useless, Thanks to find

File: src/main/java/org/folio/consortia/service/impl/UserTenantServiceImpl.java (hunk 2 above)
Comment (azizbekxm) [reply src/main/java/org/folio/consortia/service/impl/UserTenantServiceImpl.java]: Fixed

**Review by tatsiana-tarhonskaya — APPROVED** [review]

**Comment by okolawole-ebsco** [comment]
Please fill out the purpose and approach sections of the PR so that reviewers that are out of the loop may have context. Thanks!

**Comment by azizbekxm** [comment]
> Please fill out the purpose and approach sections of the PR so that reviewers that are out of the loop may have context. Thanks!

Updated, thanks

**Pull request merged**